Client/WriteApiAsync.cs

```diff
@@ -0,0 +1,337 @@
+            Arguments.CheckNonEmptyString(org, nameof(org));
+            Arguments.CheckNotNull(precision, nameof(precision));
+            
+            List<BatchWriteData> list = new List<BatchWriteData>();
+
+            foreach (var measurement in measurements)
+            {
+                var options = new BatchWriteOptions(bucket, org, precision);
+
+                BatchWriteData data = new BatchWriteMeasurement<TM>(options, _options, measurement, _measurementMapper);
+                list.Add(data);
+            }
+
+            await WriteData(org, bucket, precision, list);
+        }
+
+        /// <summary>
+        /// Write Measurements into specified bucket.
+        /// </summary>
+        /// <param name="precision">specifies the precision for the unix timestamps within the body line-protocol</param>
+        /// <param name="measurements">specifies Measurements to write into bucket</param>
+        /// <typeparam name="TM">measurement type</typeparam>
+        public async Task WriteMeasurementsAsync<TM>(WritePrecision precision, params TM[] measurements)
+        {
+            Arguments.CheckNotNull(precision, nameof(precision));
+
+            await WriteMeasurementsAsync(_options.Bucket, _options.Org, precision, measurements);
+        }
+
+        /// <summary>
+        /// Write Measurements into specified bucket.
+        /// </summary>
+        /// <param name="bucket">specifies the destination bucket for writes</param>
+        /// <param name="org">specifies the destination organization for writes</param>
+        /// <param name="precision">specifies the precision for the unix timestamps within the body line-protocol</param>
+        /// <param name="measurements">specifies Measurements to write into bucket</param>
+        /// <typeparam name="TM">measurement type</typeparam>
+        public async Task WriteMeasurementsAsync<TM>(string bucket, string org, WritePrecision precision,
+                        params TM[] measurements)
+        {
+            Arguments.CheckNonEmptyString(bucket, nameof(bucket));
+            Arguments.CheckNonEmptyString(org, nameof(org));
+            Arguments.CheckNotNull(precision, nameof(precision));
+
+            await WriteMeasurementsAsync(bucket, org, precision, measurements.ToList());
+        }
+
+        private async Task WriteData(string org, string bucket, WritePrecision precision, IEnumerable<BatchWriteData> data)
+        {
+            var sb = new StringBuilder("");
+            
+            foreach (var item in data)
+            {
+                var lineProtocol = item.ToLineProtocol();
+
+                if (string.IsNullOrEmpty(lineProtocol))
+                {
+                    continue;
+                }
+                
+                sb.Append(lineProtocol);
+                sb.Append("\n");
+            }
+            
+            if (sb.Length == 0)
+            {
+                return;
```

Comment:
Please add log message that there is nothing to write:

```suggestion
                Trace.WriteLine($"The writes: {data} doesn't contains any Line Protocol, skipping");
                return;
```